test/PoolManager.t.sol

```diff
@@ -770,6 +775,38 @@ contract PoolManagerTest is Test, Deployers, GasSnapshot {
         snapEnd();
     }
 
+    function test_swap_withFeeTakingHook() public {
+        address hookAddr = address(uint160(Hooks.AFTER_SWAP_FLAG));
+        FeeTakingHook impl = new FeeTakingHook(manager);
+        vm.etch(hookAddr, address(impl).code);
+
+        (key,) = initPoolAndAddLiquidity(currency0, currency1, IHooks(hookAddr), 100, SQRT_RATIO_1_1, ZERO_BYTES);
+
+        PoolSwapWithFeeTest router = new PoolSwapWithFeeTest(manager);
+        MockERC20(Currency.unwrap(currency0)).approve(address(router), Constants.MAX_UINT256);
+
+        IPoolManager.SwapParams memory params =
+            IPoolManager.SwapParams({zeroForOne: true, amountSpecified: 100, sqrtPriceLimitX96: SQRT_RATIO_1_2});
+        router.swapExactInput(key, params, 95);
```

Comment:
can we also add a test where the hook taking a fee causes the amountOut (after paying on behalf) to drop below minAmountOut and make sure that it reverts?